process/block/baseProcess.go

```diff
@@ -2056,7 +2057,32 @@ func createDisabledProcessDebugger() (process.Debugger, error) {
 	return debugFactory.CreateProcessDebugger(configs)
 }
 
-// WasBlockCommitted returns true if at least one block was committed
-func (bp *baseProcessor) WasBlockCommitted() bool {
-	return bp.wasBlockCommitted.IsSet()
+// NonceOfFirstCommittedBlock returns the last committed block's nonce. The optional Uint64 will contain a-not-set value
+// if no block was committed by the node
+func (bp *baseProcessor) NonceOfFirstCommittedBlock() core.OptionalUint64 {
+	bp.mutNonceOfFirstCommittedBlock.RLock()
+	defer bp.mutNonceOfFirstCommittedBlock.RUnlock()
+
+	return bp.nonceOfFirstCommittedBlock
+}
+
+func (bp *baseProcessor) setNonceOfFirstCommittedBlock(nonce uint64) {
+	// try first a read operation that will be parallelized to improve performance
```

Comment:
Taking into account that this set is called only once per round I think you can make a Lock/Unlock with defer in the beginning of this method and you can actually throw out L2074 and L2077-L2085

Reply:
Right, will remove the extra RLock-RUnlock. If we gonna use NonceOfFirstCommittedBlock extensively and performance degradation occurs, will add the RLock usage here

Reply:
Ok